### PR TITLE
[WIP] Rework Chocobo digging

### PR DESCRIPTION
**Pull request by Xaver-DaRed**
**_I affirm:_**
- [x] I have paid attention to this example and will edit again if need be to not break the formatting, or I will be ignored
- [x] I have read and understood the [Contributing Guide](https://github.com/LandSandBoat/server/blob/base/CONTRIBUTING.md)
- [NOPE] I've _**tested my code and the things my code has changed**_ since the last commit in the PR, and will test after any later commits

## What does this pull request do?

A rework of chocobo digging, based on Limit Break server implentation, code provided by Aiko.
Moved most of the work done in core to lua to minimize impact.

Changes:
- Insert changes here.

Deviations from LB implementation:
- Insert deviations here.

## Steps to test these changes
- Get a chocobo license.
- Get a bunch of greens.
- Get a fire extinguisher.
- Go digging.
- Apply fire extinguisher to PC if needed.

### CODE REVIEW

File: scripts/globals/chocobo_digging.lua [hunk 1]
```diff
@@ -801,209 +828,344 @@ local digInfo =
...
+-- Calculate skill up.
 local function calculateSkillUp(player)
     local skillRank = player:getSkillRank(xi.skill.DIG)
-    local maxSkill = utils.clamp((skillRank + 1) * 100, 0, 1000)
     local realSkill = player:getCharSkillLevel(xi.skill.DIG)
-    local increment = 1
 
-    -- this probably needs correcting
-    local roll = math.random(0, 100)
+    local digsTable =
```
Comment (TeoTwawki) [inline scripts/globals/chocobo_digging.lua]: should explain what this (and why) is while here

File: scripts/globals/chocobo_digging.lua [hunk 2]
```diff
@@ -801,209 +828,344 @@ local digInfo =
...
+        player:getCurrentMission(xi.mission.log_id.AMK) == xi.mission.id.amk.SHOCK_ARRANT_ABUSE_OF_AUTHORITY and
+        xi.amk.helpers.chocoboDig(player, zoneId, text)
+    then
+        return
+    end
 
-    else
-        local roll = math.random(0, 100)
-        local moon = VanadielMoonPhase()
+    -- Handle distance automatic fail.
```
Comment (TeoTwawki) [inline scripts/globals/chocobo_digging.lua]: my dig results from yesterday disagree with this. I was in same spot for several min, got multiple junk items. I didn't move.

**Review by TeoTwawki — COMMENTED** [review]
If intending to clean this up from the old dsp code, please make it less if/else/or/or/or/or and use some tables, index lookups, and in retail game verification.

**Comment by Xaver-DaRed** [comment]
This is still a WIP. it needs much cleaing still

**Comment by TeoTwawki** [comment]
> This is still a WIP. it needs much cleaing still

I understand its wip, but its better to give feedback as it goes isn't it? so that you don't do a bunch of work and then undo it to fix something.

File: scripts/globals/chocobo_digging.lua (hunk 2 above)
Comment (Xaver-DaRed) [reply scripts/globals/chocobo_digging.lua]: Correct.

This is custom behavior intended to reduce boting.
I will deal with it after we decide if this is something the base code could benefit from.

Comment (TeoTwawki) [reply scripts/globals/chocobo_digging.lua]: sounds like module territory then

Comment (zach2good) [reply scripts/globals/chocobo_digging.lua]: Agree with teo, it's not our place to improve on retail in the base branch, that belongs in modules

Comment (Xaver-DaRed) [reply scripts/globals/chocobo_digging.lua]: If thats the consensus, it shall be removed^^

Comment (The-Aerec) [reply scripts/globals/chocobo_digging.lua]: Can add it to an ASB module?

**Pull request closed without merging**